Pages/IndieAuth/Auth.php

```diff
@@ -61,23 +61,62 @@ function post()
             $code         = $this->getInput('code');
             $client_id    = $this->getInput('client_id');
             $redirect_uri = $this->getInput('redirect_uri');
+            
+            $headers      = self::getallheaders();
 
             $verified = Auth::verifyCode($code, $client_id, $redirect_uri);
             if ($verified['valid']) {
-                $this->setResponse(200);
-                header('Content-Type: application/x-www-form-urlencoded');
-                echo http_build_query(array(
-                    'scope'        => $verified['scope'],
-                    'me'           => $verified['me'],
-                ));
+                $this->setResponse(200);  
+                if(!empty($headers['Accept'])) {
+                    switch($headers['Accept']) {
```

Comment:
As a separate point - given headers can be uppercase or lowercase, should we handle lowercase header names?